desktop/src/dbus.rs

```diff
@@ -0,0 +1,82 @@
+//! Types and methods utilized for communicating with D-Bus
+
+use futures::StreamExt;
+use zbus::export::futures_core::Stream;
+use zbus::zvariant::{OwnedValue, Value};
+use zbus::{proxy, Connection};
+
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+#[proxy(
+    interface = "org.freedesktop.portal.Settings",
+    default_service = "org.freedesktop.portal.Desktop",
+    default_path = "/org/freedesktop/portal/desktop",
+    gen_blocking = false
+)]
+trait FreedesktopSettingsInterface {
+    fn read(&self, namespace: &str, key: &str) -> Result<OwnedValue>;
+
+    #[zbus(signal)]
+    fn setting_changed(&self, namespace: &str, key: &str, value: Value<'_>);
+}
+
+pub enum ColorScheme {
+    Default,
+    PreferLight,
+    PreferDark,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum FreedesktopSettingsError {
+    #[error("Unexpected value for color scheme: {0}")]
+    UnexpectedColorScheme(Value<'static>),
+}
+
+pub struct FreedesktopSettings<'p> {
+    proxy: FreedesktopSettingsInterfaceProxy<'p>,
+}
+
+impl<'p> FreedesktopSettings<'p> {
+    const COLOR_SCHEME: (&'static str, &'static str) =
```

Comment:
Feels like this tuple actually requires more code to unpack every time it is used.

Reply:
More code compared to what? :thinking:

Reply:
Just having two named consts.

Reply:
Unfortunately using two constants will make the code two lines longer, because their names will be longer. It's true that using a tuple requires an additional line for unpacking but it makes it possible to name those constants locally (which would require two lines when using two constants), and ensures that they are always used together (they are basically inseparable, you always use both or none).